Add ability to log items to console

HighCrit opened the pull request:
I basically have to write this everytime I do something with items. So I figured why not make a feature of it. Press INSERT (default) while hovering an item in a container or when holding an item in your main hand.

![image](https://user-images.githubusercontent.com/35771251/184888415-ea9198bb-7808-48b6-8bea-e698d8518ccd.png)

Incompleteusern commented:
Should be last commit until feedback/review.

Incompleteusern commented:
Should be last commit until feedback as the callback function is stable and everything else works.

magicus commented on src/main/java/com/wynntils/modules/questbook/overlays/ui/LootRunPage.java:
```diff
@@ -361,7 +361,11 @@ public void mouseClicked(int mouseX, int mouseY, int mouseButton) throws IOExcep
+
+                            Minecraft.getMinecraft().player.sendMessage(new TextComponentString(message));
+
+                            String startingPointMsg = "Loot run starts at [" + (int) start.getX() + ", " + (int) start.getZ() + "]";
```
This seems like an unrelated change..? It looks good, but not like it belongs here

Incompleteusern replied:
I can remove it if necessary, but It doesn't seem deserving of its own PR

magicus replied:
Okay.

magicus commented on src/main/java/com/wynntils/modules/chat/overlays/ChatOverlay.java:
```diff
@@ -272,11 +289,116 @@ private void updateLine(ChatTab tab, ITextComponent chatComponent, int updateCou
+
+    public void processQueues() {
+        for (ChatTab tab : TabManager.getAvailableTabs()) {
+            processQueue(tab);
```
I'm not sure I understand the queues fully, but my understanding is that if the quest book is not analyzed, you queue chat messages awaiting analysis to see if it should be translated. This is perhaps my only worry... 1) Are you sure questbook analysis cannot get stuck in some kind of loop/deadlock, where lack of chat feedback prevents it from finishing? And 2) this means that whenever you start playing the first chat message will be delayed until the quest book is analysed.

magicus replied:
It might be okay, I just want to point this out. An alternative is to start background analysis if it's missing, but skip trying to translate messages until the analysis is done.

Incompleteusern replied:
Its not queuing the message so much as queuing the change. It already starts background analysis and reprocesses the message later based on groupID

`
@SubscribeEvent
    public void onAnalyzeDiscoveries(QuestBookUpdateEvent.Partial e) {
        if (e.getAnalysed() == AnalysePosition.SECRET_DISCOVERIES && !ChatManager.getDiscoveriesLoaded()) {
            ChatManager.setDiscoveriesLoaded(true);
            ChatOverlay.getChat().processQueues();
        }
    } 
    `
    
    `
    case discovery:
                    if (!PlayerInfo.get(CharacterData.class).isLoaded()) {
                        translateWynnic = true;
                        translateGavellian = true;
                        break;
                    }

                    if (QuestManager.getCurrentDiscoveries().isEmpty() && !discoveriesLoaded) {
                        QuestManager.updateAnalysis(EnumSet.of(AnalysePosition.DISCOVERIES, AnalysePosition.SECRET_DISCOVERIES), true, true);
                        return new Pair<>(original, new Pair<>(ChatManager::getDiscoveriesLoaded, s -> ChatManager.processRealMessage(s).a));
                    }
                    `

magicus commented on src/main/java/com/wynntils/modules/chat/configs/ChatConfig.java:
```diff
@@ -62,30 +62,43 @@
 
-    @Setting(displayName = "Clickable Party Invites", description = "Should party invites provide a clickable command?")
+    @Setting(displayName = "Wynnic Translation Condition", description = "What is the condition for translating Wynnic and Gavellian be translated?\n\n§8Discovery - The discovery for the transcriber has been discovered\n\n§8Book - The transcriber is in the player's inventory", order = 12)
+    public TranslateConditions translateCondition = TranslateConditions.always;
```
Should it be always as default? Maybe one of the more limiting choices (i'd vote for discoveries). Otherwise it's unlikely that anyone would ever actively limit this.

Incompleteusern replied:
I don't really know what to have other than that it shouldn't be never, and some people don't really do secret discoveries especially because both of these discoveries are ultimate. And true, people wouldn't really want to limit their choices unless for RP reasons.

magicus replied:
I'm basically okay with this patch. If you think always is correct, go for it. Otherwise a quick check on discord to see if people agree..?

Incompleteusern commented:
@HeyZeer0 Can I have a review?

Incompleteusern commented:
Fixed conflicts with 1.16 preparations too

magicus commented:
I think this looks good now. I don't think we need to wait more for additional reviews.

HighCrit committed:
Add ability to log items to console

I basically have to write this everytime I do something with items. So I figured why not make a feature of it. Press INSERT (default) while hovering an item in a container or when holding an item in your main hand.

magicus reviewed (approved):
👍
